packages/gatsby-source-contentful/src/extend-node-type.js

```diff
@@ -34,10 +52,26 @@ const getBase64Image = imageProps => {
   if (!imageProps) return null
 
   const requestUrl = `https:${imageProps.baseUrl}?w=20`
-  // TODO add caching.
+
+  // Note: sha1 is unsafe for crypto but okay for this particular case
+  const shasum = crypto.createHash(`sha1`)
+  shasum.update(`requestUrl`)
+  const urlSha = shasum.digest(`hex`)
+
+  // TODO: Find the best place for this step. This is definitely not it.
+  fs.mkdirSync(CACHE_IMG_FOLDER, { recursive: true })
```

Comment:
Any reason not to use the normal cache in the non-remote case? https://www.gatsbyjs.org/docs/build-caching/#the-cache-api